### PR TITLE
Avoid warning about soon to close milestones on merged PRs

### PR DESCRIPTION
**Pull request by loremattei**
This PR fixes an issue with the `milestone` rule where the warning about having set a milestone which is close to its due date is triggered also on merges. 

Looking at [GitHub documentation](https://docs.github.com/en/rest/reference/pulls), the state of a PR is set to `closed` when a PR is closed or merged, so checking for `state == "merged"` should cover both cases.

### CODE REVIEW

File: tests/pr-milestone.test.ts [hunk 1]
```diff
@@ -145,4 +145,16 @@ describe("PR milestone checks", () => {
             expect(dm.warn).not.toHaveBeenCalled();
         })
     }
+
+    it("does not warn the milestone is closing in 4 days or less and PR is merged", async () => {
```
Comment (AliSoftware) [inline tests/pr-milestone.test.ts:149]:
```suggestion
    it("does not warn when the milestone is closing in 4 days or less but PR is merged", async () => {
```

File: tests/pr-milestone.test.ts [hunk 2]
```diff
@@ -145,4 +145,16 @@ describe("PR milestone checks", () => {
...
+        var closeDate = new Date();
+        closeDate.setDate(closeDate.getDate() + 4);
```
Comment (AliSoftware) [inline tests/pr-milestone.test.ts:151]: Just a suggestion to avoid using the value exactly at the limit value `4` that we use between "closing soon" and "not closing soon" (in risk of hitting edge case of `<` vs `≤` unexpectedly and making the test actually test "not closing soon" by mistake – even if in that case that shouldn't change anything).

```suggestion
        closeDate.setDate(closeDate.getDate() + 2);
```

Just reduces the risk of doubt when reading the test code imho; but not an issue per se, so feel free to ignore 🤷

**Review by AliSoftware — APPROVED** [review]
Thanks for the fix!
Just left a couple of nits but nothing blocking, so LGTM! 👍

**Comment by loremattei** [comment]
Thank you @AliSoftware! 🙇‍♂️  I integrated your suggestions and I'm merging this now.
